TestFoundation/TestNSOperationQueue.swift

```diff
@@ -65,4 +67,75 @@ class TestNSOperationQueue : XCTestCase {
         XCTAssertEqual(msgOperations[2], "Operation2 executed")
         XCTAssertEqual(msgOperations[3], "Operation4 executed")
     }
+
+    func test_AsyncOperation() {
+        let operation = AsyncOperation()
+        XCTAssertFalse(operation.isExecuting)
+        XCTAssertFalse(operation.isFinished)
+
+        operation.start()
+
+        while !operation.isFinished {
+            // wait until the operation finishes
+            XCTAssertTrue(operation.isExecuting)
+            XCTAssertFalse(operation.isFinished)
+        }
+
+        XCTAssertFalse(operation.isExecuting)
+        XCTAssertTrue(operation.isFinished)
+    }
+}
+
+class AsyncOperation: Operation {
+
+    private let queue = DispatchQueue(label: "async.operation.queue")
+
+    private var _executing = false
```

Comment:
Do you mean where they are updated at the end of the `queue.async()` block below?

What's the best way to do this?  `NSLock`?